### DISCUSSION TITLE
Declarative User Profiles: Support for service accounts

### DISCUSSION THREAD

**Discussion by thedmi** [keycloak/keycloak#25991] [Q&A]
### Description

While trying the declarative user profile feature, I realized that custom attributes do not show up for service account users.

Is there a specific reason why service account users are excluded from this feature? 

I think it would make sense for service account users as well. Specifically, there is a use case for this in our Keycloak setup:

We manage a few hundred device clients through the Keycloak admin console. Each such device has additional properties that are later mapped onto tokens. Currently, we maintain a custom admin theme, but this is a huge maintenance burden with each new major release. It seems that the declarative user profiles would fit nicely here, but obviously, only if the feature does support service account users.

**Comment by antikalk** [comment]
For me (on 23.0.4) it's the other way around: Service Accounts show the User Profile Attributes and their validations are also enforced.

IMO this depends highly on the Use Case: I have a different Scenario, where I need User Profile to be activated for Users and not for Service Accounts. The users have custom attributes that are verified, e.g. phoneNumber, birthdate which I don't need for the service account users that are only technical.

Unfortunately on the API when updating Service Accounts the User Profile attributes are validated and enforced, not allowing me to update a service account without providing a phoneNumber, birthdate etc...

I would love to have the possibility to configure User Profile for Users and Service Accounts independently and/or have a config to set whether an attribute should also apply to service account users.

**Comment by codebude** [comment]
Hi @thedmi @antikalk ,

We are using 24.0.4 and experiencing the same problem. Have you found a solution in the meantime?

**Reply by antikalk** [reply]
The issue I had (that service accounts require the user profile attributes) is fixed with KC24. Afaik there is still no way of enforcing or configuring user profile attributes for service-accounts...

**Reply by ahbonsu** [reply]
just migrated to 24.0.4 from 23.x and facing the same issue. I mean, the attributes are there (I can see then when querying kc) but unfortunately they are not shown on the web-ui :(
